### ISSUE TITLE
Not work with procure model

### ISSUE THREAD

**Issue by Olegsplit** [ytti/oxidized#1436]
In new version(0.24) stop work with Procure model.
In version 0.21 worked fine.

I changed ssh.rb from old version and all works fine.

Debug with new ssh.rb:
```
DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
DEBUG -- : lib/oxidized/job.rb: Starting fetching process for DEVICE_NAME at 2018-07-06 12:14:04 UTC
DEBUG -- : AUTH METHODS::["none", "publickey", "password"]
DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to DEVICE_NAME
DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
WARN -- : DEVICE_IP raised IOError with msg "closed stream"
DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for DEVICE_NAME
DEBUG -- : DEVICE_IP raised Errno::ECONNREFUSED with msg "Connection refused - connect(2) for "DEVICE_IP" port 23"
DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for DEVICE_NAME
DEBUG -- : lib/oxidized/job.rb: Config fetched for DEVICE_NAME at 2018-07-06 12:14:07 UTC
WARN -- : DEVICE_NAME status no_connection, retry attempt 1

```

**Comment by ytti** [comment]
Can you try to add 'keyboard-interactive' auth method.
https://github.com/ytti/oxidized/blob/master/docs/Configuration.md#ssh-auth-methods

**Comment by Olegsplit** [comment]
Not work

**Comment by ytti** [comment]
And you are saying if you run 0.24 with only ssh.rb from 0.21 it works? Can you do input log (input:\ndebug: true) for the failing session, so we can see what the session looks.

**Comment by Olegsplit** [comment]
D, [2018-07-10T12:15:00.205508 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:01.205990 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:01.206592 #4445] DEBUG -- : lib/oxidized/worker.rb: Added DEVICE_NAME to the job queue
D, [2018-07-10T12:15:01.206742 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:01.206937 #4445] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for DEVICE_NAME at 2018-07-10 09:15:01 UTC
D, [2018-07-10T12:15:01.207352 #4445] DEBUG -- : AUTH METHODS::["none", "publickey", "password", "keyboard-interactive"]
D, [2018-07-10T12:15:01.207853 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to DEVICE_NAME
D, [2018-07-10T12:15:02.181254 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:02.207023 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:03.207330 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-10T12:15:03.988343 #4445]  WARN -- : DEVICE_IP raised IOError with msg "closed stream"
D, [2018-07-10T12:15:03.988636 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for DEVICE_NAME
D, [2018-07-10T12:15:03.989996 #4445] DEBUG -- : DEVICE_IP raised Errno::ECONNREFUSED with msg "Connection refused - connect(2) for "DEVICE_IP" port 23"
D, [2018-07-10T12:15:03.990196 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for DEVICE_NAME
D, [2018-07-10T12:15:03.990384 #4445] DEBUG -- : lib/oxidized/job.rb: Config fetched for DEVICE_NAME at 2018-07-10 09:15:03 UTC
W, [2018-07-10T12:15:04.208669 #4445]  WARN -- : DEVICE_NAME status no_connection, retry attempt 1
D, [2018-07-10T12:15:04.209029 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:04.209359 #4445] DEBUG -- : lib/oxidized/worker.rb: Added DEVICE_NAME to the job queue
D, [2018-07-10T12:15:04.209516 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:04.209718 #4445] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for DEVICE_NAME at 2018-07-10 09:15:04 UTC
D, [2018-07-10T12:15:04.210072 #4445] DEBUG -- : AUTH METHODS::["none", "publickey", "password", "keyboard-interactive"]
D, [2018-07-10T12:15:04.210291 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to DEVICE_NAME
D, [2018-07-10T12:15:05.011359 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:05.209831 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:06.210303 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-10T12:15:07.018924 #4445]  WARN -- : DEVICE_IP raised IOError with msg "closed stream"
D, [2018-07-10T12:15:07.019200 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for DEVICE_NAME
D, [2018-07-10T12:15:07.028343 #4445] DEBUG -- : DEVICE_IP raised Errno::ECONNREFUSED with msg "Connection refused - connect(2) for "DEVICE_IP" port 23"
D, [2018-07-10T12:15:07.028559 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for DEVICE_NAME
D, [2018-07-10T12:15:07.028748 #4445] DEBUG -- : lib/oxidized/job.rb: Config fetched for DEVICE_NAME at 2018-07-10 09:15:07 UTC
W, [2018-07-10T12:15:07.210919 #4445]  WARN -- : DEVICE_NAME status no_connection, retry attempt 2
D, [2018-07-10T12:15:07.211246 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:07.211537 #4445] DEBUG -- : lib/oxidized/worker.rb: Added DEVICE_NAME to the job queue
D, [2018-07-10T12:15:07.211694 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:07.211942 #4445] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for DEVICE_NAME at 2018-07-10 09:15:07 UTC
D, [2018-07-10T12:15:07.212369 #4445] DEBUG -- : AUTH METHODS::["none", "publickey", "password", "keyboard-interactive"]
D, [2018-07-10T12:15:07.212615 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to DEVICE_NAME
D, [2018-07-10T12:15:08.094770 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:08.212033 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:09.213358 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:09.797908 #4445] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login commands at DEVICE_NAME
D, [2018-07-10T12:15:09.798206 #4445] DEBUG -- : lib/oxidized/input/cli.rb: Running post_login command: "no page", block: nil at DEVICE_NAME
D, [2018-07-10T12:15:09.798375 #4445] DEBUG -- : lib/oxidized/input/ssh.rb no page @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:09.798679 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:10.200410 #4445] DEBUG -- : lib/oxidized/model/model.rb Collecting commands' outputs
D, [2018-07-10T12:15:10.200742 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show version
D, [2018-07-10T12:15:10.200906 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show version @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:10.201176 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:10.213801 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:10.603015 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show modules
D, [2018-07-10T12:15:10.603357 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show modules @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:10.603678 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:11.005495 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show interfaces transceiver
D, [2018-07-10T12:15:11.005884 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show interfaces transceiver @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:11.006241 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:11.214128 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:11.407863 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show flash
D, [2018-07-10T12:15:11.408177 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show flash @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:11.408501 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:12.105357 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show system-information
D, [2018-07-10T12:15:12.105654 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show system-information @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:12.105948 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:12.214565 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:12.507890 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show system information
D, [2018-07-10T12:15:12.508193 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show system information @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:12.508535 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:13.111399 #4445] DEBUG -- : lib/oxidized/model/model.rb Executing show running-config
D, [2018-07-10T12:15:13.111726 #4445] DEBUG -- : lib/oxidized/input/ssh.rb show running-config @ DEVICE_NAME with expect: /^\r?([\w.-]+# )$/
D, [2018-07-10T12:15:13.112044 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:13.214958 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:14.216138 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-10T12:15:14.618265 #4445]  WARN -- : DEVICE_IP raised IOError with msg "closed stream"
D, [2018-07-10T12:15:14.618556 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for DEVICE_NAME
D, [2018-07-10T12:15:14.620383 #4445] DEBUG -- : DEVICE_IP raised Errno::ECONNREFUSED with msg "Connection refused - connect(2) for "DEVICE_IP" port 23"
D, [2018-07-10T12:15:14.620592 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for DEVICE_NAME
D, [2018-07-10T12:15:14.620871 #4445] DEBUG -- : lib/oxidized/job.rb: Config fetched for DEVICE_NAME at 2018-07-10 09:15:14 UTC
W, [2018-07-10T12:15:15.216810 #4445]  WARN -- : DEVICE_NAME status no_connection, retry attempt 3
D, [2018-07-10T12:15:15.217118 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:15.217437 #4445] DEBUG -- : lib/oxidized/worker.rb: Added DEVICE_NAME to the job queue
D, [2018-07-10T12:15:15.217585 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:15.217777 #4445] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for DEVICE_NAME at 2018-07-10 09:15:15 UTC
D, [2018-07-10T12:15:15.218143 #4445] DEBUG -- : AUTH METHODS::["none", "publickey", "password", "keyboard-interactive"]
D, [2018-07-10T12:15:15.218348 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to DEVICE_NAME
D, [2018-07-10T12:15:16.104797 #4445] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at DEVICE_NAME
D, [2018-07-10T12:15:16.217914 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-10T12:15:17.218399 #4445] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-10T12:15:18.112719 #4445]  WARN -- : DEVICE_IP raised IOError with msg "closed stream"
D, [2018-07-10T12:15:18.112972 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for DEVICE_NAME
D, [2018-07-10T12:15:18.189997 #4445] DEBUG -- : DEVICE_IP raised Errno::ECONNREFUSED with msg "Connection refused - connect(2) for "DEVICE_IP" port 23"
D, [2018-07-10T12:15:18.190172 #4445] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for DEVICE_NAME
D, [2018-07-10T12:15:18.190369 #4445] DEBUG -- : lib/oxidized/job.rb: Config fetched for DEVICE_NAME at 2018-07-10 09:15:18 UTC
D, [2018-07-10T12:15:18.219089 #4445] DEBUG -- : Exec: Execute: "/home/oxidized/.config/oxidized/extra/oxidized-report-git-commits ~/gitdir/ | mail -s \"Oxidized updates for ${OX_NODE_NAME}\" mail@mail.com"
W, [2018-07-10T12:15:18.220502 #4445]  WARN -- : DEVICE_NAME status no_connection, retries exhausted, giving up
D, [2018-07-10T12:15:18.220660 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 1 of 1
D, [2018-07-10T12:15:18.220867 #4445] DEBUG -- : lib/oxidized/worker.rb: Running :nodes_done hook
D, [2018-07-10T12:15:19.221203 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:20.221743 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-10T12:15:21.222267 #4445] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1

**Comment by ytti** [comment]
No, not this file, the ```ìp-ssh``` file which is generated if input: debug: true is set

**Comment by Olegsplit** [comment]
In device log I see that oxidized_user logined to device.
I 07/10/18 12:15:02.05 03362 auth: User 'oxidized_user' logged in from
            IP_OXIDIZED to SSH  session
I 07/10/18 12:15:03.13 00179 mgr: SME SSH from IP_OXIDIZED - MANAGER Mode
I 07/10/18 12:15:03.31 03363 auth: User 'oxidized_user' logged out of SSH
            session from  IP_OXIDIZED
I 07/10/18 12:15:04.88 03362 auth: User 'oxidized_user' logged in from
            IP_OXIDIZED to SSH  session
I 07/10/18 12:15:06.24 00179 mgr: SME SSH from IP_OXIDIZED - MANAGER Mode
I 07/10/18 12:15:06.39 03363 auth: User 'oxidized_user' logged out of SSH
            session from  IP_OXIDIZED
I 07/10/18 12:15:07.97 03362 auth: User 'oxidized_user' logged in from
            IP_OXIDIZED to SSH  session
I 07/10/18 12:15:09.40 00179 mgr: SME SSH from IP_OXIDIZED - MANAGER Mode
I 07/10/18 12:15:14.12 03363 auth: User 'oxidized_user' logged out of SSH
            session from  IP_OXIDIZED
I 07/10/18 12:15:15.98 03362 auth: User 'oxidized_user' logged in from
            IP_OXIDIZED to SSH  session
I 07/10/18 12:15:17.30 00179 mgr: SME SSH from IP_OXIDIZED - MANAGER Mode
I 07/10/18 12:15:17.48 03363 auth: User 'oxidized_user' logged out of SSH
            session from  IP_OXIDIZED

**Comment by ytti** [comment]
We need the file called '<ipaddres>-ssh', which is generated if:
```
input:
  debug: true
```

**Comment by Olegsplit** [comment]
[IP-ssh.txt](https://github.com/ytti/oxidized/files/2180034/IP-ssh.txt)

**Comment by ytti** [comment]
Ok seems like it gets bunch of commands, but for some reason after running-configuration, won't output anything.

I really don't see any meaningful difference in ```ssh.rb``` https://github.com/ytti/oxidized/compare/0.21.0...0.24.0#diff-2acf2a59874e4200ef75364597cdb52c

Are you saying you can use 0.24.0 just fine, as long as you get ssh.rb from 0.21.0? Because model difference would be probably  better explanation. Are you using input: debug: true? If so, does disabling it help?

If above does not move us forward, I'll need sanitised access to affected device to test against.

**Comment by Olegsplit** [comment]
```
- Are you saying you can use 0.24.0 just fine, as long as you get ssh.rb from 0.21.0?
- Yes.
```

```
- Are you using input: debug: true?
- Yes
```

```
- If so, does disabling it help?
- The same.
```

```
- I'll need sanitised access to affected device to test against.
- What is mean?
```

**Comment by ytti** [comment]
It means I don't know what's wrong, and I've run out of easy things to propose, so I'd like to take a look at myself. So if possible, please setup sanitised device I can access from 91.198.120.1 and send me authentication details to saku@ytti.fi and I'll figure out what is wrong.

**Comment by WiXZlo** [comment]
Hello.
I have same problem after upgrade from 0.22 to 0.24. I tested oxidized with different input methods and got results:
1. ssh.rb: 
- v0.21 - OK
- v0.22 - error
- v0.23 - error
- v0.24 - error
2. telnet.rb:
- v0.21 - OK
- v0.22 - OK
- v0.23 - error
- v0.24 - error

With telnet v0.23 I got warning in log:
```
D, [2018-07-19T17:39:09.417768 #29271] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-19T17:39:09.656643 #29271]  WARN -- : 172.17.17.33 raised Oxidized::PromptUndetect with msg "unable to detect prompt: (?-mix:^\r?([\w.-]+# )$)"
D, [2018-07-19T17:39:09.656797 #29271] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for hp2620-24-p-1
```
Device log `cat -v ~/.config/oxidized/logs/172.17.17.33-telnet`
```
^[[2J^[[?7l^[[3;23r^[[?6l^[[1;1H^[[?25l^[[1;1HHP J9625A 2620-24-PoEP Switch^M
Software revision RA.15.16.0006^M
^M
Copyright (C) 1991-2015 Hewlett-Packard Development Company, L.P.

                   RESTRICTED RIGHTS LEGEND
 Confidential computer software.  Valid license from HP required for possession,
 use or copying. Consistent with FAR 12.211 and 12.212, Commercial Computer
 Software, Computer Software Documentation, and Technical Data for Commercial
 Items are licensed to the U.S. Government under vendor's standard commercial
 license.
                   HEWLETT-PACKARD DEVELOPMENT COMPANY, L.P.
                   20555 State Highway 249, Houston, TX 77070

We'd like to keep you up to date about:
  * Software feature updates
  * New product announcements
  * Special events
Please register your products now at:  www.hp.com/networking/register^[[1;20r^[[1;1H^[[1;24r^[[20;1H^[[24;1HUsername: ^[[?25h^[[24;1H^[[?25h^[[24;11H^[[24;11H^[[?25h^[[24;1
```

Log from version v0.22 and SSH input method
```
I, [2018-07-19T17:53:19.854907 #29650]  INFO -- : Oxidized starting, running as pid 29650
I, [2018-07-19T17:53:19.855396 #29650]  INFO -- : lib/oxidized/nodes.rb: Loading nodes
D, [2018-07-19T17:53:19.855504 #29650] DEBUG -- : resolving DNS for hp2620-24-p-1...
D, [2018-07-19T17:53:19.855531 #29650] DEBUG -- : IPADDR 172.17.17.33
D, [2018-07-19T17:53:19.855621 #29650] DEBUG -- : node.rb: resolving node key 'model', with passed global value of '' and node value 'procurve'
D, [2018-07-19T17:53:19.855648 #29650] DEBUG -- : node.rb: setting node key 'model' to value 'junos' from global
D, [2018-07-19T17:53:19.855672 #29650] DEBUG -- : node.rb: returning node key 'model' with value 'procurve'
D, [2018-07-19T17:53:19.855687 #29650] DEBUG -- : lib/oxidized/node.rb: Loading model "procurve"
D, [2018-07-19T17:53:19.856192 #29650] DEBUG -- : lib/oxidized/model/model.rb Added all to the commands list
D, [2018-07-19T17:53:19.856228 #29650] DEBUG -- : lib/oxidized/model/model.rb Added secret to the commands list
D, [2018-07-19T17:53:19.856245 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show version to the commands list
D, [2018-07-19T17:53:19.856259 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show modules to the commands list
D, [2018-07-19T17:53:19.856271 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show system power-supply to the commands list
D, [2018-07-19T17:53:19.856283 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show interfaces transceiver to the commands list
D, [2018-07-19T17:53:19.856294 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show flash to the commands list
D, [2018-07-19T17:53:19.856304 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show system-information to the commands list
D, [2018-07-19T17:53:19.856319 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show system information to the commands list
D, [2018-07-19T17:53:19.856330 #29650] DEBUG -- : lib/oxidized/model/model.rb Added show running-config to the commands list
D, [2018-07-19T17:53:19.856580 #29650] DEBUG -- : node.rb: resolving node key 'input', with passed global value of 'ssh' and node value ''
D, [2018-07-19T17:53:19.856612 #29650] DEBUG -- : node.rb: returning node key 'input' with value 'ssh'
D, [2018-07-19T17:53:19.905697 #29650] DEBUG -- : node.rb: resolving node key 'output', with passed global value of 'git' and node value ''
D, [2018-07-19T17:53:19.905767 #29650] DEBUG -- : node.rb: returning node key 'output' with value 'git'
D, [2018-07-19T17:53:19.913537 #29650] DEBUG -- : node.rb: resolving node key 'username', with passed global value of '' and node value ''
D, [2018-07-19T17:53:19.913579 #29650] DEBUG -- : node.rb: setting node key 'username' to value 'USERNAME' from global
D, [2018-07-19T17:53:19.913620 #29650] DEBUG -- : node.rb: returning node key 'username' with value 'USERNAME'
D, [2018-07-19T17:53:19.913637 #29650] DEBUG -- : node.rb: resolving node key 'password', with passed global value of '' and node value ''
D, [2018-07-19T17:53:19.913653 #29650] DEBUG -- : node.rb: setting node key 'password' to value 'PASSWORD' from global
D, [2018-07-19T17:53:19.913668 #29650] DEBUG -- : node.rb: returning node key 'password' with value 'PASSWORD'
I, [2018-07-19T17:53:19.913722 #29650]  INFO -- : lib/oxidized/nodes.rb: Loaded 1 nodes
D, [2018-07-19T17:53:20.213573 #29650] DEBUG -- : lib/oxidized/core.rb: Starting the worker...
D, [2018-07-19T17:53:20.213678 #29650] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-19T17:53:20.213779 #29650] DEBUG -- : lib/oxidized/worker.rb: Added hp2620-24-p-1 to the job queue
D, [2018-07-19T17:53:20.213808 #29650] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-19T17:53:20.213885 #29650] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for hp2620-24-p-1 at 2018-07-19 14:53:20 UTC
D, [2018-07-19T17:53:20.214135 #29650] DEBUG -- : AUTH METHODS::["none", "publickey", "password"]
D, [2018-07-19T17:53:20.214194 #29650] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to hp2620-24-p-1
D, [2018-07-19T17:53:20.893524 #29650] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at hp2620-24-p-1
D, [2018-07-19T17:53:21.213962 #29650] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-19T17:53:22.214199 #29650] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-19T17:53:22.699752 #29650]  WARN -- : 172.17.17.33 raised IOError with msg "closed stream"
D, [2018-07-19T17:53:22.699844 #29650] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for hp2620-24-p-1
D, [2018-07-19T17:53:22.699884 #29650] DEBUG -- : lib/oxidized/job.rb: Config fetched for hp2620-24-p-1 at 2018-07-19 14:53:22 UTC
W, [2018-07-19T17:53:23.214694 #29650]  WARN -- : hp2620-24-p-1 status no_connection, retry attempt 1
D, [2018-07-19T17:53:23.214809 #29650] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
D, [2018-07-19T17:53:23.215037 #29650] DEBUG -- : lib/oxidized/worker.rb: Added hp2620-24-p-1 to the job queue
D, [2018-07-19T17:53:23.215060 #29650] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-19T17:53:23.215103 #29650] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for hp2620-24-p-1 at 2018-07-19 14:53:23 UTC
D, [2018-07-19T17:53:23.215316 #29650] DEBUG -- : AUTH METHODS::["none", "publickey", "password"]
D, [2018-07-19T17:53:23.215379 #29650] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to hp2620-24-p-1
D, [2018-07-19T17:53:24.084824 #29650] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/^\r?([\w.-]+# )$/] at hp2620-24-p-1
D, [2018-07-19T17:53:24.215251 #29650] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2018-07-19T17:53:25.215528 #29650] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
W, [2018-07-19T17:53:25.997048 #29650]  WARN -- : 172.17.17.33 raised IOError with msg "closed stream"
D, [2018-07-19T17:53:25.997164 #29650] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for hp2620-24-p-1
D, [2018-07-19T17:53:25.997207 #29650] DEBUG -- : lib/oxidized/job.rb: Config fetched for hp2620-24-p-1 at 2018-07-19 14:53:25 UTC
W, [2018-07-19T17:53:26.216440 #29650]  WARN -- : hp2620-24-p-1 status no_connection, retries exhausted, giving up
D, [2018-07-19T17:53:26.216561 #29650] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 1 of 1
D, [2018-07-19T17:53:26.216606 #29650] DEBUG -- : lib/oxidized/worker.rb: Running :nodes_done hook
D, [2018-07-19T17:53:27.216795 #29650] DEBUG -- : lib/oxidized/worker.rb: Jobs running: 0 of 1 - ended: 0 of 1
```
Device log `cat -v ~/.config/oxidized/logs/172.17.17.33-ssh`
```
HP J9625A 2620-24-PoEP Switch^M^M
Software revision RA.15.16.0006^M^M
^M^M
Copyright (C) 1991-2015 Hewlett-Packard Development Company, L.P.^M
^M
                   RESTRICTED RIGHTS LEGEND^M
 Confidential computer software.  Valid license from HP required for possession,^M
 use or copying. Consistent with FAR 12.211 and 12.212, Commercial Computer^M
 Software, Computer Software Documentation, and Technical Data for Commercial^M
 Items are licensed to the U.S. Government under vendor's standard commercial^M
 license.^M
                   HEWLETT-PACKARD DEVELOPMENT COMPANY, L.P. ^M
                   20555 State Highway 249, Houston, TX 77070^M
^M
^[[1;15r^[[1;1H^[[24;1HPress any key to continue^[[15;1H^[[?25h^[[24;27H
```

**Comment by philippreston** [comment]
Seeing this issue as well.  Resolved by using 0.2.1

**Comment by frank42hh** [comment]
Hi,
I seem to have the same problem as well, with the current version 0.28.0, but not with every procurve switch.

I have 2 switches, both are `HP J9145A 2910al-24G Switch`, one with Software revision W.15.14.0016 and one with Software revision W.15.14.0009. As far as I can see, the expect output is the same for both systems, and I checked IP authorized managers, which is also the same for both systems.
I can login via ssh on both systems just fine, and for both systems I see the login from oxidized in their event log.
But, on one system I get the expected output from procurve.rb, and on the other one I get this debug output:

`D, [2021-12-07T17:53:51.187413 #12657] DEBUG -- : resolving DNS for hostname.domain.tld...
D, [2021-12-07T17:55:49.463970 #12657] DEBUG -- : lib/oxidized/worker.rb: Added /hostname.domain.tld to the job queue
D, [2021-12-07T17:55:49.464151 #12657] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for hostname.domain.tld at 2021-12-07 16:55:49 UTC
D, [2021-12-07T17:55:49.464807 #12657] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to hostname.domain.tld
D, [2021-12-07T17:55:49.466264 #12657] DEBUG -- net.ssh.transport.session[2af429004ef0]: establishing connection to 10.7.23.115:22
D, [2021-12-07T17:55:50.580665 #12657] DEBUG -- : lib/oxidized/input/ssh.rb: expecting [/(^\r|\e\[24;[0-9][hH])?([\w\s.-]+[#>] )($|(\e\[24;[0-9][0-9]?[hH]){3})/] at hostname.domain.tld
I, [2021-12-07T17:55:52.290348 #12657]  INFO -- net.ssh.connection.session[2af428dc60a8]: 10.7.23.115 delete channel 0 which closed locally and remotely
W, [2021-12-07T17:55:52.391181 #12657]  WARN -- : 10.7.23.115 raised IOError with msg "closed stream"
D, [2021-12-07T17:55:52.391245 #12657] DEBUG -- : lib/oxidized/node.rb: Oxidized::SSH failed for hostname.domain.tld
D, [2021-12-07T17:55:52.391295 #12657] DEBUG -- : lib/oxidized/job.rb: Config fetched for hostname.domain.tld at 2021-12-07 16:55:52 UTC
W, [2021-12-07T17:55:52.465179 #12657]  WARN -- : /hostname.domain.tld status no_connection, retries exhausted, giving up
`
I have no clue how to resolve this. any ideas?
